ivy/stateful/module.py

```diff
@@ -817,6 +817,16 @@ def register_buffer(self, var_name, value):
         """Set the buffer at any place within the class."""
         self._set_buffers({var_name: value})
 
+    def eval(self, mode: bool = False):
+        # disables training mode for child modules
+        # if it exists
+        if hasattr(self, "training"):
+            self.training = mode
+        for module in self.v:
```

Comment:
`self.v` is a `ivy.Container` with `ivy.Array` leaves right? Not sure why it would have an `ivy.Module` instance. Did you intend to use `self.__dict__` instead?

Reply:
I am iterating the keys, which are going to be Module instances. Iterating over `self.__dict__` will be suboptimal, as we need to just focus on the Module instances that are contributing to weight updates.

Reply:
yeah you're right, thanks for clarifying!